node.go

```diff
@@ -514,6 +520,20 @@ func (n *Node) switchToCandidate() {
 	n.switchState(CANDIDATE)
 }
 
+// Execute `f` in a separate go routine, but ensures that functions
+// are executed in order.
+func (n *Node) serializeGoRoutine(nextCh *(chan bool), f func()) {
```

Comment:
The goal is to not block the node when posting a state change (or error). The default StateChange handler is channel based, and unbuffered (not a solution anyway). If the user is not dequeuing, the node could be blocked.
Options are for the node to have a single go routine but use an array or list of events that are added and notify the central go routine doing the dispatching.
I feel that in this case, the proposed solution is elegant and simpler. I don't think we can expect thousands of changes a second ;-). But we do want changes to be added to the handler in order. What do you think?